eth_utils/address.py

```diff
@@ -188,18 +182,21 @@ def to_checksum_address(address):
         )
         for i in range(2, 42)
     ))
-    return checksum_address
+    if is_bytes(checksum_address):
+        return to_text(primitive=checksum_address)
+    else:
+        return to_text(text=checksum_address)
 
 
-@coerce_args_to_text
-def is_checksum_address(value):
+def is_checksum_address(primitive=None, hexstr=None, text=None):
+    value = to_text(primitive, hexstr, text)
     if not is_hex_address(value):
         return False
-    return value == to_checksum_address(value)
+    return value == to_checksum_address(text=value)
 
 
-@coerce_args_to_text
-def is_checksum_formatted_address(value):
+def is_checksum_formatted_address(primitive=None, hexstr=None, text=None):
```

Comment:
I think this still needs to maintain a single `value` parameter as the only argument.